pulsar-common/src/main/java/org/apache/pulsar/common/nar/NarUnpacker.java

```diff
@@ -87,6 +92,7 @@ static File doUnpackNar(final File nar, final File baseWorkingDirectory, Runnabl
                  FileLock lock = channel.lock()) {
                 File narWorkingDirectory = new File(parentDirectory, md5Sum);
                 if (narWorkingDirectory.mkdir()) {
+                    setPosixFilePermissions(narWorkingDirectory,"rwxr-x---");
```

Comment:
And please check code format.

Reply:
Thank for your review. Not only mkdirs/mkdir can create new directory which need set permission, there are some other method can create new file such as FileOutputStream if file do not exist first.So I think we need to create file then set some special permission such as w permission for file before FileOutputStream.I also think like you that a new method should be created to create the file or directory. I will try to solve it.Thanks again!